src/util-base64.c

```diff
@@ -156,9 +156,25 @@ Base64Ecode DecodeBase64(uint8_t *dest, uint32_t dest_size, const uint8_t *src,
         }
     }
 
-    if (!valid && mode == BASE64_MODE_RFC4648) {
-        padding = B64_BLOCK - bbidx;
-        *decoded_bytes += ASCII_BLOCK - padding;
+    if (bbidx > 0 &&
+            ((!valid && mode == BASE64_MODE_RFC4648) || (valid && mode == BASE64_MODE_RFC2045))) {
+        padding = bbidx > 1 ? B64_BLOCK - bbidx
+                            : 2; // Decoded bytes for 1 or 2 base64 encoded bytes is 1
+        uint32_t numDecoded_blk = ASCII_BLOCK - (padding < B64_BLOCK ? padding : ASCII_BLOCK);
+        if (dest_size < *decoded_bytes + numDecoded_blk) {
+            SCLogDebug("Destination buffer full");
+            ecode = BASE64_ECODE_BUF;
+            *consumed_bytes += leading_sp;
+            SCLogNotice("consumed_bytes: %d, decoded_bytes: %d", *consumed_bytes, *decoded_bytes);
+            return ecode;
+        }
+        if (dest_size % 3 !=
```

Comment:
what is the logic of having the destination size be a multiple of 3?

Reply:
It should not be less than 3. This is wrong.

Reply:
ok. I'm thinking about how to handle this in the callers.